flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -826,6 +836,19 @@ private Optional<Object> getRawValue(String key, boolean canBePrefixMap) {
         }
     }
 
+    /**
+     * This method will do the following steps to get the value of a config option:
+     *
+     * <p>1. get the value from {@link Configuration}. <br>
+     * 2. if key is not found, try to get the value with fallback keys from {@link Configuration}
+     * <br>
+     * 3. if no fallback keys are found, return {@link Optional#empty()}.
+     *
+     * <p>Current logic is not able to get the default value of the fallback key's ConfigOption, if
+     * the given ConfigOption has no default value.
```

Comment:
we can mark this as a note although I think this is by design, we get option value by option key firstly and then *try to * fallback to fallback key which is used when  user only configured fallback key. The note is useful when users upgrade from an old flink version which only owned the fallback keys to the higher flink version which deprecated these fallback keys.